src/Layouts/Partials/FillsPanel.vala

```diff
@@ -91,16 +92,25 @@ public class Akira.Layouts.Partials.FillsPanel : Gtk.Grid {
     private void create_event_bindings () {
         toggled = false;
         window.event_bus.selected_items_changed.connect (on_selected_items_changed);
+
         window.event_bus.fill_deleted.connect (() => {
             add_btn.show ();
             window.main_window.left_sidebar.queue_resize ();
         });
+
         add_btn.clicked.connect (() => {
-            list_model.add_fill.begin (selected_item);
+            var model_item = create_model ();
+            list_model.add_item.begin (model_item);
             selected_item.reset_colors ();
             add_btn.hide ();
             window.main_window.left_sidebar.queue_resize ();
         });
+
+        list_model.items_changed.connect ((position, removed, added) => {
+            if (selected_item != null) {
+                selected_item.has_fill = (added == 1);
```

Comment:
Maybbe adding a comment to clarify what this code actually does would make it better. 

For me it's not super easy to get the meaning of this at a first glance, so maybe adding some hints would help our future self to understand what's happening. 

What do you think?

Reply:
Good point.